components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java

```diff
@@ -325,10 +325,13 @@ public boolean doPostAuthenticate(String userName, boolean authenticated,
                         String emailTemplate = null;
                         int tenantId = userStoreManager.getTenantId();
                         String firstName = null;
+                        String userStoreDomain = userStoreManager.getRealmConfiguration()
+                                .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+                        String domainSpecificUserName = UserCoreUtil.addDomainToName(userName, userStoreDomain);
+
                         try {
-                            firstName =
-                                    Utils.getClaimFromUserStoreManager(userName, tenantId,
-                                            "http://wso2.org/claims/givenname");
+                            firstName = Utils.getClaimFromUserStoreManager(domainSpecificUserName, tenantId,
+                                    "http://wso2.org/claims/givenname");
```

Comment:
There are no constant values defined for "http://wso2.org/claims/givenname" inside the org.wso2.carbon.identity.mgt and it is defined inside org.wso2.carbon.identity.mgt.endpoint so to avoid an unnecessary export in pom file, I haven't used the constant value for this URL.